lib/bolt/plugin/aws.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Aws
+      class EC2
+        attr_accessor :client
+        attr_reader :config
+
+        def initialize(config)
+          require 'aws-sdk-ec2'
+          @config = config
+          @logger = Logging.logger[self]
+        end
+
+        def name
+          'aws::ec2'
+        end
+
+        def hooks
+          %w[inventory_targets]
+        end
+
+        def config_client(opts)
+          return client if client
+
+          options = {}
+
+          if opts.key?('region')
+            options[:region] = opts['region']
+          end
+          if opts.key?('profile')
+            options[:profile] = opts['profile']
+          end
+          if config.aws&.key?('credentials')
+            creds = File.expand_path(config.aws['credentials'])
+            if File.exist?(creds)
+              options[:credentials] = ::Aws::SharedCredentials.new(path: creds)
+            else
+              @logger.warn("Cannot load credentials file #{config.aws['credentials']}")
+            end
+          end
+
+          ::Aws::EC2::Client.new(options)
+        end
+
+        def inventory_targets(opts)
+          client = config_client(opts)
+          resource = ::Aws::EC2::Resource.new(client: client)
+
+          # Retrieve a list of EC2 instances and create a list of targets
+          # Note: It doesn't seem possible to filter stubbed responses...
+          resource.instances(filters: opts['filters']).map do |instance|
+            next unless instance.state.name == 'running'
+            target = {}
+
+            if opts.key?('uri')
+              uri = lookup(instance, opts['uri'])
+              target['uri'] = uri if uri
+            end
+            if opts.key?('name')
+              real_name = lookup(instance, opts['name'])
+              target['name'] = real_name if real_name
+            end
+            if opts.key?('config')
+              target['config'] = resolve_config(instance, opts['config'])
+            end
+
+            target if target['uri'] || target['name']
+          end.compact
+        end
+
+        # Look for an instance property specified in the inventory file
+        # All instance properties are prefixed with 'instance.'
+        # If a property is not found, then a warning is displayed
+        # If a property is not an instance property, it's left as-is
+        def lookup(instance, property, default = nil)
+          if default.nil? || property.start_with?("instance.")
```

Comment:
I remember you asking about this and I'm not sure why past-me said to keep `instance`, but I think you can have users just put the data name they want without `instance.` and not need this. It adds risk of `instance` being misspelled or something, and just makes the user type more unnecessarily. Sorry if I misled you!